File: 1/1000s/14s/1432/MaxDifferenceYouCanGetFromChangingAnInteger.py
# https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/description/?envType=daily-question&envId=2025-06-15

class Solution:
    def maxDiff(self, num: int) -> int:
        numAsString=str(num)
        lengthOfString=len(numAsString)
        maxNumToBeFormed=-1*(1e9+10)
        minNumToBeFormed=1e9+10
        for i in range(lengthOfString-1, -1, -1):
            currentChar=numAsString[i]
            for i2 in range(0, 10, 1):
                newStringToBeFormed=''
                for i3 in range(0, lengthOfString, 1):
                    if numAsString[i3]!=currentChar:
                        newStringToBeFormed+=numAsString[i3]
                    else:
                        newStringToBeFormed+=str(i2)
                valToCheck=int(newStringToBeFormed)
                currentCharToCheck=newStringToBeFormed[0]
                if currentCharToCheck!='0':
                    maxNumToBeFormed=max(maxNumToBeFormed, valToCheck)
                    minNumToBeFormed=min(minNumToBeFormed, valToCheck)
        return maxNumToBeFormed-minNumToBeFormed

File: 1/1000s/14s/1432/MaxDifferenceYouCanGetFromChangingAnInteger.java
/*
https://leetcode.com/problems/max-difference-you-can-get-from-changing-an-integer/description/?envType=daily-question&envId=2025-06-15
*/

class Solution {
    public int maxDiff(int num) {
        String numAsString=String.valueOf(num);
        StringBuilder toBeProcessedString=new StringBuilder(numAsString);
        int lengthOfString=numAsString.length(),
            maxNumToBeFormed=Integer.MIN_VALUE,
            minNumToBeFormed=Integer.MAX_VALUE;
        for(int i=lengthOfString-1; i>=0; --i){
            char currentChar=toBeProcessedString.charAt(i);
            for(int i2=0; i2<=9; ++i2){
                StringBuilder newStringToBeFormed=new StringBuilder();
                for(int i3=0; i3<lengthOfString; ++i3){
                    if(toBeProcessedString.charAt(i3)!=currentChar){
                        newStringToBeFormed.append(toBeProcessedString.charAt(i3));
                    } else{
                        newStringToBeFormed.append(String.valueOf(i2));
                    }
                }
                int valToCheck=Integer.parseInt(String.valueOf(newStringToBeFormed));
                char currentCharToCheck=newStringToBeFormed.charAt(0);
                if(currentCharToCheck!='0'){
                    maxNumToBeFormed=Math.max(maxNumToBeFormed, valToCheck);
                    minNumToBeFormed=Math.min(minNumToBeFormed, valToCheck);
                }
            }
        }
        return maxNumToBeFormed-minNumToBeFormed;
    }
}